File: TestRomanNumerals/Convertor1.cs
using System.Collections.Generic;
using System.Text;

namespace CSharpRomanNumerals
{
    public static class Convertor
    {


        public static string AsRomanNumeral(int number)
        {
            var values = new int[] { 1000, 900, 500, 400, 100, 90, 50, 40, 10,9, 5, 4, 1 };
            var symbols = new string[] { "M", "CM", "D", "CD", "C", "XC", "L", "XL", "X", "IX", "V", "IV", "I" };
            var result = new StringBuilder();
            for (int i = 0; i < values.Length; i++)
            {
                var value = values[i];
                while (number >= value)
                {
                    result.Append(symbols[i]);
                    number -= value;
                }
            }
            return result.ToString();
        }
    }
}

File: Roman Numerals/RomanNumeralsVB/Convertor1.vb
Module Convertor
    Function AsRomanNumeral(ByVal number As Integer) As String
        Dim values = New Integer() {1000, 900, 500, 400, 100, 90, 50, 40, 10, 9, 5, 4, 1}
        Dim symbols = New String() {"M", "CM", "D", "CD", "C", "XC", "L", "XL", "X", "IX", "V", "IV", "I"}
        Dim result = New StringBuilder()

        For i As Integer = 0 To values.Length - 1
            Dim value = values(i)

            While number >= value
                result.Append(symbols(i))
                number -= value
            End While
        Next

        Return result.ToString()
    End Function
End Module
